src/aws-cpp-sdk-core/include/smithy/tracing/TracingUtils.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+
+#include <aws/core/monitoring/HttpClientMetrics.h>
+#include <smithy/Smithy_EXPORTS.h>
+#include <smithy/tracing/Meter.h>
+#include <functional>
+#include <chrono>
+#include <utility>
+
+namespace smithy {
+    namespace components {
+        namespace tracing {
+            static const char SMITHY_METRICS_DNS_DURATION[] = "smithy.client.http.dns_duration";
```

Comment:
in order to avoid a creation of these string literals in each translation unit that includes that header
I suggest to move these string literals to `TracingUtils` class as a static members
then initialize them in a corresponding .cpp (which I guess you tried to avoid, but this results in duplication of this literals in the resulting binary)